src/GitHub.VisualStudio/TeamExplorer/Connect/GitHubConnectSection.cs

```diff
@@ -156,11 +256,59 @@ public void Login()
             StartFlow(UIControllerFlow.Authentication);
         }
 
+        public void OpenRepository()
+        {
+            var old = Repositories.FirstOrDefault(x => String.Equals(Holder.ActiveRepo?.RepositoryPath, x.LocalPath, StringComparison.CurrentCultureIgnoreCase));
+            // open the solution selection dialog when the user wants to switch to a different repo
+            // since there's no other way of changing the source control context in VS
+            if (SelectedRepository != old)
```

Comment:
I haven't thought this through, but this is a reference equality check. Should this be:

```csharp
if (!Equals(SelectedRepository, old))
```